src/Microsoft.Health.Dicom.SqlServer/Features/Workitem/WorkitemSqlQueryGenerator.cs

```diff
@@ -0,0 +1,179 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Linq;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag;
+using Microsoft.Health.Dicom.Core.Features.Query;
+using Microsoft.Health.Dicom.Core.Features.Query.Model;
+using Microsoft.Health.Dicom.SqlServer.Features.ExtendedQueryTag;
+using Microsoft.Health.Dicom.SqlServer.Features.Query;
+using Microsoft.Health.Dicom.SqlServer.Features.Schema;
+using Microsoft.Health.Dicom.SqlServer.Features.Schema.Model;
+using Microsoft.Health.SqlServer;
+using Microsoft.Health.SqlServer.Features.Schema.Model;
+using Microsoft.Health.SqlServer.Features.Storage;
+
+namespace Microsoft.Health.Dicom.SqlServer.Features.Workitem
+{
+    internal class WorkitemSqlQueryGenerator : BaseSqlQueryGenerator
+    {
+        private readonly IndentedStringBuilder _stringBuilder;
+        private readonly QueryExpression _queryExpression;
+        private readonly SchemaVersion _schemaVersion;
+        private const string WorkitemTableAlias = "w";
+
+        public WorkitemSqlQueryGenerator(
+            IndentedStringBuilder stringBuilder,
+            QueryExpression queryExpression,
+            SqlQueryParameterManager sqlQueryParameterManager,
+            SchemaVersion schemaVersion,
+            int partitionKey)
+            : base(stringBuilder, sqlQueryParameterManager, partitionKey)
+        {
+            _stringBuilder = stringBuilder;
+            _queryExpression = queryExpression;
+            _schemaVersion = schemaVersion;
+
+            if ((int)_schemaVersion < SchemaVersionConstants.SupportUpsRsSchemaVersion)
+            {
+                throw new BadRequestException(DicomSqlServerResource.SchemaVersionNeedsToBeUpgraded);
+            }
+
+            Build();
+        }
+
+        private void Build()
+        {
+            string projectionTableAlias = "f";
+
+            AppendSelect(projectionTableAlias);
+
+            AppendFilterTable(projectionTableAlias);
+
+            AppendOptionRecompile();
+        }
+
+        private void AppendFilterTable(string filterAlias)
+        {
+            _stringBuilder
+                .AppendLine("( SELECT ")
+                .Append(VLatest.Workitem.WorkitemUid, WorkitemTableAlias).AppendLine(",")
+                .Append(VLatest.Workitem.WorkitemKey, WorkitemTableAlias).AppendLine()
+                .AppendLine($"FROM {VLatest.Workitem.TableName} {WorkitemTableAlias}");
+
+            AppendExtendedQueryTagTables();
```

Comment:
Can we name the new methods differently? Maybe AppendLongSchemaQueryTagTables()? Whatever name we choose to represent.

Reply:
done